### ISSUE TITLE
Azure AD OAUTH Support

### ISSUE THREAD

**Issue by michaelsatish** [kubeshop/testkube#4347]
Any plans to support Azure AD as a OAUTH provider? 

There is support for only GitHub.
https://github.com/kubeshop/testkube/tree/develop/pkg/oauth

My company mandates Azure to be the only ODIC and OAUTH provider.

**Comment by vsukhin** [comment]
hey @michaelsatish please ping @TheBrunoLopes we have a lot of diffferent Auth providers in our Enterprise edition @dejanzele

**Comment by michaelsatish** [comment]
Thank you @vsukhin 

At this point in time, I don't have a need to have more than one environment, and the reason to go with open source.

**Comment by michaelsatish** [comment]
@vsukhin It would be informative if the documentation https://docs.testkube.io/ had a section to showcase additional options offered in enterprise version. 

On a different note, is AWS S3 artifact storage without minio is an enterprise version offering? I see only minio is supported 
https://github.com/kubeshop/testkube/tree/develop/pkg/storage

**Comment by TheBrunoLopes** [comment]
Hey @michaelsatish thank you for your feedback.
Definitely a good idea to add the additional enterprise options in our documentation. we will work on it

Regarding your minio question. It seems we use it as a proxy to connect to all different types of s3 compatible
storages https://docs.testkube.io/articles/deploying-in-aws/#examples-of-aws-s3-bucket-configuration

The Testkube Enterprise solution works in a slightly different way when it comes to the storage system which may allow us to have some flexibility. I would love to understand your needs more. Is using Minio an issue for your use case ?

**Comment by michaelsatish** [comment]
@TheBrunoLopes Thank for sharing the link. We scan images for vulnerabilities periodically, and if found, we upgrade to the newer version if there is a fix. Thats why I have minion outside of testkube, so I can keep up with the upstream project. 

If testkube can directly interact with AWS S3, there will be one less moving part and more secure at least in my use case. If Stack Overflow survey is anything to go by, AWS is leading the cloud adoption and supporting AWS S3 should help many. 

https://survey.stackoverflow.co/2023/#section-most-popular-technologies-cloud-platforms

**Comment by vsukhin** [comment]
yes, Testkube uses s3 api to store logs and test artifacts, it's connected to MinIo by default, but you can connect it directly to aws s3 bucket

**Comment by michaelsatish** [comment]
Wonderful. @vsukhin Thank you. I did go through the below link, may be I am not getting it, probably the case.
https://docs.testkube.io/articles/deploying-in-aws#examples-of-aws-s3-bucket-configuration

```yaml
minio:
      enabled: true # required to be able to access AWS S3 (minio is used as a proxy)
```

Is testkube using minio storage implementation code as a proxy without a minio deployment?

**Comment by vsukhin** [comment]
Guess @nicufk did it without minio
